activation/handler.go

```diff
@@ -50,6 +51,9 @@ type Handler struct {
 	fetcher         system.Fetcher
 	poetCfg         PoetConfig
 
+	signerMtx sync.Mutex
+	signers   map[types.NodeID]*signing.EdSigner
```

Comment:
You only need to store the node IDs as far as I can tell. How about just
```suggestion
	nodeIdsMtx sync.Mutex
	nodeIds    map[types.NodeID]struct{}
```

Reply:
I wanted to keep it in line with how multi-smesher support is implemented in other components and how it was suggested on the first issue: https://github.com/spacemeshos/go-spacemesh/issues/5085

I will still need to pass `*signing.EdSigner` in `Register` to satisfy this interface and then it makes little difference to keep an empty object or a pointer to an instance of `signing.EdSigner` that is used anyway in other components.

Wdyt?

Reply:
That proposal only talks about `Register(*signing.EdSigner)` function. It makes little difference memory-wise, but that's not about that. It's about redundant information. You don't need to store the signers (they are useless in the context of the Handler) and if they are stored, then other devs will wonder what for.

Anyway - not a big deal, I leave it up to you to decide :)